poll-webapp/src/main/webapp/vue-app/poll-activity-stream-extension/components/CreatePollToolbarAction.vue

```diff
@@ -0,0 +1,143 @@
+<!--
+This file is part of the Meeds project (https://meeds.io/).
+ 
+Copyright (C) 2022 Meeds Association contact@meeds.io
+ 
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU Lesser General Public
+License as published by the Free Software Foundation; either
+version 3 of the License, or (at your option) any later version.
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+Lesser General Public License for more details.
+ 
+You should have received a copy of the GNU Lesser General Public License
+along with this program; if not, write to the Free Software Foundation,
+Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+-->
+<template>
+  <v-btn
+    :ripple="false"
+    class="d-flex flex-row align-center py-2"
+    text
+    @click="openCreatePollDrawer">
+    <v-icon
+      color="amber darken-1"
+      size="27">
+      fa-poll
+    </v-icon>
+    <v-span class="body-2 font-weight-bold ms-5 mt-1 dark-grey-color">
+      {{ $t('poll.title') }}
+    </v-span>
+  </v-btn>
+</template>
+<script>
+export default {
+  data() {
+    return {
+      pollAction: 'create',
+      savedPoll: {}
+    };
+  },
+  props: {
+    activityId: {
+      type: String,
+      default: null,
+    },
+    message: {
+      type: String,
+      default: null,
+    },
+    maxMessageLength: {
+      type: Number,
+      default: 0,
+    },
+    templateParams: {
+      type: Object,
+      default: null,
+    },
+    files: {
+      type: Array,
+      default: null,
+    },
+    activityType: {
+      type: Array,
+      default: null,
+    },
```

Comment:
Please change all those props into data since it will not be provided by a parent Vue component

Reply:
All these props are passed into the  "templateParams" object in the social module when using the extension registry for poll and kudos.

Reply:
Ah, ok, it should then be defined before the `data`, see it here https://eslint.vuejs.org/rules/order-in-components.html

Besides, you are logging the error instead of display a message alert to the user when it happens `this.$root.$emit('alert-message'...`
+
You are changing a props using `this.activityType.push('poll');` which is not recommended.

Reply:
fixed!